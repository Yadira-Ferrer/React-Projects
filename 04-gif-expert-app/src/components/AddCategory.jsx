import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    /* No recargar la página */
    event.preventDefault();
    /* No enviar cadenas vacias */
    if (inputValue.trim().length <= 1) return;
    /* Añadir la nueva categoria */
    onNewCategory(inputValue.trim());
    /* Limpiar input */
    setInputValue('');
  };

  return (
    <div className="form-container">
      <form onSubmit={onSubmit} aria-label="form">
        <input
          type="text"
          placeholder="Buscar Gifs"
          value={inputValue}
          onChange={onInputChange}
        />
      </form>
    </div>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
