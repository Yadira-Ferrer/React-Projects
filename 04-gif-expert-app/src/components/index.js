export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';

/* Archivo de barril */
