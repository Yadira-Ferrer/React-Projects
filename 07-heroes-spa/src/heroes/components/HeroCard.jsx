import { Link } from 'react-router-dom';

const CharactersByHero = ({ alter_ego, characters }) => {
  /* if (alter_ego === characters) return <></>;
  return <p className="">{characters}</p>; */
  return alter_ego === characters ? <></> : <p className="">{characters}</p>;
};

export const HeroCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters,
}) => {
  const heroImageUrl = `/heroes/${id}.jpg`;
  //const charactersByHero = <p className="">{characters}</p>;

  return (
    <div className="col animate__animated animate__fadeIn">
      <div className="card">
        <div className="row no-gutters">
          <div className="col-4">
            <img
              src={heroImageUrl}
              className="img-fluid rounded-start"
              alt={superhero}
              style={{ height: '100%' }}
            />
          </div>

          <div className="col-8">
            <div className="card-body">
              <h5 className="card-title">{superhero}</h5>
              <p className="card-text">{alter_ego}</p>
              {/* {alter_ego !== characters && charactersByHero} */}
              <CharactersByHero characters={characters} alter_ego={alter_ego} />
              <p className="card-text">
                <small className="text-muted">{first_appearance}</small>
              </p>
              <Link className="text-decoration-none" to={`/hero/${id}`}>
                MÃ¡s ...
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
