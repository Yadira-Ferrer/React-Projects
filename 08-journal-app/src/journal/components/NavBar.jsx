import { MenuOutlined, PowerSettingsNewRounded } from '@mui/icons-material';
import { AppBar, Toolbar, IconButton, Grid, Typography } from '@mui/material';

/* eslint-disable react/prop-types */ // TODO: upgrade to latest eslint tooling
export const NavBar = ({ drawerWidth = 240 }) => {
  return (
    <AppBar
      position="fixed"
      sx={{
        width: { sm: `calc(100% - ${drawerWidth}px)` },
        ml: { sm: `${drawerWidth}px` },
      }}
    >
      <Toolbar>
        <IconButton
          color="inheri"
          edge="start"
          sx={{ mr: 2, display: { sm: 'none' } }}
        >
          <MenuOutlined />
        </IconButton>

        <Grid
          container
          direction="row"
          justifyContent="space-between"
          alignItems="center"
        >
          <Typography variant="h6" noWrap component="div">
            {' '}
            JournalApp{' '}
          </Typography>
          <IconButton color="error">
            <PowerSettingsNewRounded />
          </IconButton>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};
