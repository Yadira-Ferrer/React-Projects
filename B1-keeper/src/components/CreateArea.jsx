import { useForm } from '../hooks/useForm';
import AddIcon from '@mui/icons-material/Add';
import { Fab, Zoom } from '@mui/material';
import { useState } from 'react';

export const CreateArea = ({ onNewNote }) => {
  const [isExpanded, setExpanded] = useState(false);

  const { title, content, onInputChange, onResetForm } = useForm({
    title: '',
    content: '',
    size: '',
  });

  const onFormSubmit = (event) => {
    event.preventDefault();

    if (title.length < 1 && content.length < 1) return;

    const newNote = {
      id: new Date().getTime(),
      title,
      content,
      size:
        content.length < 135
          ? 'small'
          : content.length < 250
          ? 'medium'
          : 'large',
    };

    onNewNote && onNewNote(newNote);
    onResetForm();
  };

  const expand = () => {
    setExpanded(true);
  };

  return (
    <form className="create-note" onSubmit={onFormSubmit}>
      {isExpanded && (
        <input
          className="input-title"
          name="title"
          placeholder="Title"
          type="text"
          value={title}
          onChange={onInputChange}
        />
      )}
      <textarea
        name="content"
        onClick={expand}
        value={content}
        onChange={onInputChange}
        placeholder="Take a note..."
        rows={isExpanded ? 3 : 1}
      ></textarea>
      {isExpanded && (
        <Fab type="submit">
          <AddIcon />
        </Fab>
      )}
    </form>
  );
};
