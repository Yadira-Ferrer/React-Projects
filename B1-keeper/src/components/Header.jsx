import LightbulbIcon from '@mui/icons-material/Lightbulb';

export const Header = () => {
  return (
    <header>
      <h1>
        <LightbulbIcon /> Keeper
      </h1>
    </header>
  );
};
