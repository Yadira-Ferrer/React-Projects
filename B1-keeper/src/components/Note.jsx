export const Note = ({ title, content, notesize }) => {
  return (
    <div className={`note note_${notesize}`}>
      <h1>{title}</h1>
      <p>{content}</p>
    </div>
  );
};
