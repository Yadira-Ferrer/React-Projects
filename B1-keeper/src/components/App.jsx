import { useNotes } from '../hooks/useNotes';
import { Footer } from './Footer';
import { Header } from './Header';

/* import notes from '../notes'; */
import { CreateArea } from './CreateArea';
import { NoteList } from './NoteList';

function App() {
  const { notes, handleNewNote, handleDeleteNote } = useNotes();

  return (
    <div>
      <Header />
      <CreateArea onNewNote={handleNewNote} />
      <NoteList notes={notes} onDeleteNote={handleDeleteNote} />
      <Footer />
    </div>
  );
}

export default App;
