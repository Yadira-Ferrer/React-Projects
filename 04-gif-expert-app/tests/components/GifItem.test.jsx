import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {
  const title = 'Rugrats';
  const url = 'https://rugrats.com/rugrats.jpg';

  test('Debe coincidir con el Shapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug();
    //expect(screen.getByRole('img').src).toBe(url);

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe mostrar el tÃ­tulo en el componente', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
