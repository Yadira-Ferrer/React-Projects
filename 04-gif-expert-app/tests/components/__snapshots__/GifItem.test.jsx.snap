// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe coincidir con el Shapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Rugrats"
      src="https://rugrats.com/rugrats.jpg"
    />
    <p>
      Rugrats
    </p>
  </div>
</div>
`;
